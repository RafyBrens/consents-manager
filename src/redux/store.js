import { combineReducers, createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { ConsentReducer } from './consent';

/* Create root reducer, containing all features of the application */
const rootReducer = combineReducers({
  consent: ConsentReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(compose(thunk)))
);

export default store;
