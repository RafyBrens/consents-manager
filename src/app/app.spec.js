import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import toJson from 'enzyme-to-json';
import store from '../redux/store';
import App from './app';

it('renders without crashing', () => {
  const wrapper = mount(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
