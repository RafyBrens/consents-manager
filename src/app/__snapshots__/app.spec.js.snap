// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <app>
    <div
      className="app"
    >
      <Counter>
        <div>
          <div>
            Base component
          </div>
          <button
            onClick={[Function]}
            type="button"
          >
            Click me!
          </button>
          <div>
            You clicked: 
            <strong>
              0
            </strong>
             times.
          </div>
        </div>
      </Counter>
    </div>
  </app>
</Provider>
`;
